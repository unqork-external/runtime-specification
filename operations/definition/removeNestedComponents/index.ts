export * from './removeNestedComponents.model'
export * from './removeNestedComponents.options'
