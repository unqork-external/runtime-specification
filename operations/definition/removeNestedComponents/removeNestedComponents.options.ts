import { CollectionOf, Description, Default } from '@tsed/schema'

import { TargetedOperationOptions } from '../../interface'

export class RemoveNestedComponentsOperationOptions extends TargetedOperationOptions {
  @Description('Key of the nestable in which the new components should be removed.')
  nestable: string

  @Default([])
  @Description('Ids of the nestable components to  be removed.')
  @CollectionOf(String)
  nestablesToRemove?: string[] = []

  @Default(false)
  @Description('Optional parameter. If true, any components currently in the named nestable will be removed.')
  removeAllChildren?: boolean = false
}
