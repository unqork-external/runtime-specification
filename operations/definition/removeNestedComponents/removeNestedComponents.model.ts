import { DiscriminatorValue, Required } from '@tsed/schema'

import { RemoveNestedComponentsOperationOptions } from './removeNestedComponents.options'
import { Alpha } from '../../../../decorators'
import { AutogenDisabled } from '../../decorators/autogenDisabled.decorator'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.REMOVE_NESTED_COMPONENTS)
@Alpha()
@AutogenDisabled()
export class RemoveNestedComponentsOperation extends Operation<OperationTypes.REMOVE_NESTED_COMPONENTS> {
  public type = OperationTypes.REMOVE_NESTED_COMPONENTS as const

  @Required()
  public options: RemoveNestedComponentsOperationOptions = new RemoveNestedComponentsOperationOptions()
}
