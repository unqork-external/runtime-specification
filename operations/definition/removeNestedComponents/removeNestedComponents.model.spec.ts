import { RemoveNestedComponentsOperation } from './removeNestedComponents.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: RemoveNestedComponentsOperation Tests', () => {
  let schema
  beforeAll(() => {
    const response = generateSchemaAndValidate(RemoveNestedComponentsOperation)
    schema = response.schema
  })

  it('should match this JSON Schema', () => {
    expect(schema).toMatchSnapshot()
  })
})
