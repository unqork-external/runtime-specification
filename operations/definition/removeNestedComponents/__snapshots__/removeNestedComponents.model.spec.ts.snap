// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: RemoveNestedComponentsOperation Tests should match this JSON Schema 1`] = `
{
  "autogenDisabled": true,
  "definitions": {
    "RemoveNestedComponentsOperationOptions": {
      "properties": {
        "nestable": {
          "description": "Key of the nestable in which the new components should be removed.",
          "type": "string",
        },
        "nestablesToRemove": {
          "description": "Ids of the nestable components to  be removed.",
          "items": {
            "default": [],
            "type": "string",
          },
          "type": "array",
        },
        "removeAllChildren": {
          "default": false,
          "description": "Optional parameter. If true, any components currently in the named nestable will be removed.",
          "type": "boolean",
        },
        "shouldExecute": {
          "default": true,
          "description": "
    Conditionally execute this operation.
    Can set to false to temporarily disabled when debugging.
  ",
          "type": [
            "string",
            "boolean",
          ],
        },
        "targetKey": {
          "description": "Key of the component for which we want to remove the nested component",
          "type": "string",
        },
      },
      "type": "object",
    },
  },
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/RemoveNestedComponentsOperationOptions",
    },
    "type": {
      "const": "REMOVE_NESTED_COMPONENTS",
      "examples": [
        "REMOVE_NESTED_COMPONENTS",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
